javaparser-core/src/main/java/com/github/javaparser/ast/expr/InstanceOfExpr.java

```diff
@@ -50,15 +53,22 @@
 
     private ReferenceType type;
 
+    @OptionalProperty
+    private SimpleName name;
+
     public InstanceOfExpr() {
         this(null, new NameExpr(), new ClassOrInterfaceType());
     }
 
-    @AllFieldsConstructor
     public InstanceOfExpr(final Expression expression, final ReferenceType type) {
         this(null, expression, type);
```

Comment:
In all the other nodes, all the constructors call the most specific constructor to avoid constructor spaghetti. That would mean the this-call gets "null" at the end for example.

Reply:
Done :)